crates/holochain/src/core/ribosome/guest_callback/init.rs

```diff
@@ -317,24 +317,6 @@ mod slow_tests {
         );
     }
 
-    #[tokio::test(flavor = "multi_thread")]
-    async fn test_init_implemented_invalid_return() {
```

Comment:
I guess we have to remove this because we can no longer build a wasm that has an invalid return type?

It's a shame to have to lose a test that checks that Holochain correctly handles the invalid return type. This is really a regression test for something that is still possible, just not with the latest HDK.

Any ideas for what we could do to keep this tested?

Reply:
There's definitely value in keeping these runtime checks

Instead of creating invalid Wasm that won't compile, what do you think of constructing the invalid return data directly then testing against it?

Reply:
I'm not sure I follow the plan there. This is a way of testing what the conductor does with bad return values without WASMs?

Reply:
Not sure, hence why I asked, but on looking at the previous tests I don't think so. An idea I think we could implement is making `hdk_extern` support an additional / hidden/ internal / for testing only attribute that would act as an escape hatch to skip the compile time return type checks i.e `#[hdk_extern(__danger_skip_compile_checks)]`. Though I am not sure whether this would defeat the purpose of the compile time checks that have been put in place. What do you think?

Reply:
Could we use inline zomes? Those wouldn't have their return type checked by WASM compilation so that might be a solution?

Otherwise, is there are more direct way of testing the code that handles the hook calls? Perhaps we can refactor a bit to extract the code that deserializes the response based on the function name so we can write a test for that without needing a zome at all.

I think if we can avoid adding test utilities to the `hdk_derive` then I'd prefer we don't do that. It's complex enough code without needing to test it with multiple feature flags.

Reply:
Great idea, I had not considered inline zomes, I think they would work great for this usecase. I'll have a look at this.

> Otherwise, is there are more direct way of testing the code that handles the hook calls? Perhaps we can refactor a bit to extract the code that deserializes the response based on the function name so we can write a test for that without needing a zome at all.

I have noticed some like [this one](https://github.com/holochain/holochain/blob/b55f6c0a8fd91ed7ded3f48c3155940fd8091aad/crates/holochain/src/conductor/conductor/tests.rs#L822), but this one seems to test something different. The test cases that check for invalid return types are currently in the test wasms. I think moving them to inline zomes would be better with the new compile time checks.